dataactbroker/handlers/fileHandler.py

```diff
@@ -406,19 +418,22 @@ def uploadFile(self):
 
     def generateD1File(self):
         """ Initiates the generation of D1 """
+        jobDb = self.interfaces.jobDb
         requestDict = RequestDictionary(self.request)
 
         if not (requestDict.exists("submission_id") and requestDict.exists("start") and requestDict.exists("end")):
             exc = ResponseException("Generate D1 Files route requires submission_id, start, and end", StatusCode.CLIENT_ERROR)
-            return JsonResponse.create(exc, exc.status)
+            return JsonResponse.error(exc, exc.status)
 
         submission_id = requestDict.getValue("submission_id")
         start_date = requestDict.getValue("start")
         end_date = requestDict.getValue("end")
-
+        job = jobDb.getJobBySubmissionFileTypeAndJobType(submission_id, self.EXTERNAL_FILE_TYPE_MAP["D1"], "file_upload")
+        job.start_date = datetime.strptime(start_date,"%m/%d/%Y").date()
+        job.end_date = datetime.strptime(end_date,"%m/%d/%Y").date()
```

Comment:
The frontend has users pick from a calendar that sends the dates in that format, so there is no free entry.  If a bad format is sent, this will raise an exception and return a 500.  We could catch that and return a 400 with a more generic message if that would be preferable.